reporter/reporter.go

```diff
@@ -73,6 +74,8 @@ func Report(ctx context.Context, err error) error {
 
 	if r, ok := FromContext(ctx); ok {
 		return r.Report(ctx, e)
+	} else {
+		return errors.New("No reporter in provided context.")
```

Comment:
I'd agree. This could break some applications, but I think it's better to fail loudly here.